### PR TITLE
[impv] Change the parameter types of task and workflow to Union[timedelta, int]

### LINKED ISSUE #122: The timeout types of workflow and task parameters are inconsistent
The workflow parameter timeout is of type int, but the task parameter timeout is timedelta.

And the parameter timeout of the task cannot support defining tasks in yaml for the current type timedelta.

image

image

I plan to set the timeout type to Union [timedelta, int], which not only unifies the timeout type but also conveniently supports defining tasks in Python and YAML

### PR DESCRIPTION
**Pull request by lukeyan2023**
## Purpose of the pull request

close #122 

## Brief Summary of The Change

Change the parameter types of task and workflow to Union[timedelta, int]

## Pull Request checklist

I confirm that the following checklist has been completed.

- [ ] Add/Change **test cases** for the changes.
- [ ] Add/Change the related **documentation**, should also change `docs/source/config.rst` when you change file `default_config.yaml`.
- [ ] (Optional) Add your change to `UPDATING.md` when it is an incompatible change.

### CODE REVIEW

File: src/pydolphinscheduler/core/task.py [hunk 1]
```diff
@@ -245,13 +245,22 @@ def workflow(self, workflow: Optional[Workflow]):
...
         self._workflow = workflow
 
+    @staticmethod
+    def _parse_timeout(val: Any) -> Any:
+        if val is None or isinstance(val, timedelta):
+            return timedelta2timeout(val) if val else 0
+        elif val >= 0:
+            return val
+        else:
+            raise PyDSParamException("The timeout value cannot be negative")
```
Comment (zhongjiajie) [inline src/pydolphinscheduler/core/task.py:255]: Maybe we can embed this logic into function `timeout`

**Review by zhongjiajie — COMMENTED** [review]
Thanks for your contributions, I left some comments on this PR

BTW, could you format the code by running `tox -e auto-lint` to fit our code style?

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks for your contribution

**Comment by lukeyan2023** [comment]
> LGTM, thanks for your contribution

Thank you very much for your help. I didn't have time to check the news for something a few days ago

**Comment by zhongjiajie** [comment]
You're welcome, you could continue to contribute if you are still interested in this project. Just create some PR for this repo

**Comment by lukeyan2023** [comment]
> You're welcome, you could continue to contribute if you are still interested in this project. Just create some PR for this repo

I plan to contribute seasunel tasks, haha. I need more advice from you

**Pull request merged**